lib/Adapter/TolerantParser/Refactor/TolerantImportName.php

```diff
@@ -97,35 +102,42 @@ private function assertNotAlreadyImported(Node $node, NameImport $nameImport): v
         }
     }
 
-    private function currentClassIsSameAsImportClass(Node $node, FullyQualifiedName $className): bool
-    {
-        if (!$node instanceof ClassDeclaration) {
-            return false;
-        }
-
-        if ((string) $node->getNamespacedName() === (string) $className) {
-            return true;
-        }
-
-        return false;
-    }
-
     /**
      * @param NameImport $nameImport
      * @param array<ResolvedName> $imports
-     * @return ResolvedName
+     * @return array|null
      */
-    private function findExistingName(NameImport $nameImport, array $imports): ResolvedName
+    private function findExistingImport(NameImport $nameImport, array $imports): ?array
     {
         $nameImportParts = $nameImport->name()->toArray();
 
-        foreach ($imports as $import) {
+        foreach ($imports as $name => $import) {
             if ($import->getNameParts() === $nameImportParts) {
-                return $import;
+                // fqn already used in imports
+                return [$name, $import];
             }
         }
 
-        return $imports[$nameImport->name()->head()->__toString()];
+        $shortName = $nameImport->name()->head()->__toString();
+        if (array_key_exists($shortName, $imports)) {
+            // short name already used in imports
+            return [$shortName, $imports[$shortName]];
+        }
+
+        return null;
+    }
+
+    private function currentClassIsSameAsImportClass(Node $node, FullyQualifiedName $className): bool
+    {
+        if (!$node instanceof ClassDeclaration) {
```

Comment:
what about interfaces and traits? I think you can check for the interface providing `getNamespacedName`, or I think there is a marker class `ClassLIke` or similar

Reply:
I didn't change that code. Git just shuffled the view around.

I will check that anyway.

Reply:
Oh yeah, looks suspicious anyway :smile:

Reply:
I am now checking for the `ClassLike` and `NamespacedNameInterface` interfaces. Latter is required for `getNamespacedName()`.